.github/workflows/tests.yml

```diff
@@ -15,22 +15,36 @@ jobs:
     runs-on: ubuntu-latest
     steps:
 
-      - name: Set up Go 1.19
+      - name: Set up bootstrap Go 1.20
         uses: actions/setup-go@v5.0.0
         with:
-          go-version: 1.19
+          go-version: '1.20'
+          cache: false
         id: go
 
       - name: Check out code into the Go module directory
         uses: actions/checkout@v3.0.2
 
+      - name: Cache downloaded Go version
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/sdk
+            ~/go/bin/go*
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            ~/.cache/golangci-lint/
+          key: ${{ runner.os }}-go-sdk-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-sdk
+
```

Comment:
I disabled cache for linter and go actions and this now serves as a common cache for everything including the downloaded SDK (Go installation). This will resolve all the "tar overwriting" problems I was running into during upgrade. Should be also faster.